src/NHibernate/Transaction/AdoTransaction.cs

```diff
@@ -186,6 +186,13 @@ public void Commit()
 
 				log.Debug("Start Commit");
 
+				foreach (var sharingSession in session.ConnectionManager.SessionsSharingManager)
```

Comment:
It serves for transaction scope too. It would be strange to involve the `AdoTransaction` for transaction scope handling. And at each transaction renewal, we would have to pass that list from the old to the new.

I am renaming that list list `DependentSessions`.